crates/ruff_linter/src/rules/ruff/rules/property_without_return.rs

```diff
@@ -0,0 +1,119 @@
+use ruff_macros::{ViolationMetadata, derive_message_formats};
+use ruff_python_ast::identifier::Identifier;
+use ruff_python_ast::visitor::{Visitor, walk_expr, walk_stmt};
+use ruff_python_ast::{Expr, Stmt, StmtFunctionDef};
+use ruff_python_semantic::analyze::{function_type, visibility};
+
+use crate::checkers::ast::Checker;
+use crate::{FixAvailability, Violation};
+
+/// ## What it does
+/// Detects class `@property` methods that does not have a `return` statement.
+///
+/// ## Why is this bad?
+/// Property methods are expected to return a computed value, a missing return in a property usually indicates an implementation mistake.
+///
+/// ## Example
+/// ```python
+/// class User:
+///     @property
+///     def full_name(self):
+///         f"{self.first_name} {self.last_name}"
+/// ```
+///
+/// Use instead:
+/// ```python
+/// class User:
+///     @property
+///     def full_name(self):
+///         return f"{self.first_name} {self.last_name}"
+/// ```
+///
+/// ## References
+/// - [Python documentation: The property class](https://docs.python.org/3/library/functions.html#property)
+#[derive(ViolationMetadata)]
+#[violation_metadata(preview_since = "0.14.7")]
+pub(crate) struct PropertyWithoutReturn {
+    name: String,
+}
+
+impl Violation for PropertyWithoutReturn {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::None;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let Self { name } = self;
+        format!("{name} is a property without a `return` statement")
```

Comment:
I think we prefer to put the name in backticks for clarity:

```suggestion
        format!("`{name}` is a property without a `return` statement")
```